### PR TITLE
feat: allow unset blocklist clients

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/586.

## Changes

* ~~Move the config file logic out of the `config/mod.rs` file and into a `config/settings.rs` file (I was really hoping that the diff would be nice here).~~ I renamed it back.
* Allow the blocklist client to be unspecified.
* Made the blocklist client endpoint a `url::Url`.
* Add the block observer to the main join.

## Testing Information

Added a unit test for when the blocklist client is set via an environment variable, and manually tested what happens if the stanza in the config is set, but it is currently unset.

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

**Review by cylewitruk — APPROVED** [review]
Looks good

File: signer/src/main.rs [hunk 1]
```diff
@@ -78,6 +78,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         // Our global termination signal watcher. This does not run using `run_checked`
         // as it sends its own shutdown signal.
         run_shutdown_signal_watcher(context.clone()),
+        run_checked(run_block_observer, &context),
         // The rest of our services which run concurrently, and must all be
         // running for the signer to be operational.
```
Comment (matteojug) [inline signer/src/main.rs:83]: nit: is this `run_checked` expected to be above the comment below?

File: signer/src/transaction_signer.rs [hunk 2]
```diff
@@ -657,6 +650,14 @@ where
         Ok(())
     }
 
+    async fn check_blocklist(&self, address: &str) -> bool {
```
Comment (matteojug) [inline signer/src/transaction_signer.rs:653]: nit: the name can be confusing wrt the return value (eg, does true mean the blocklist wanna block this?); what about something like `(check_)?blocklist_accept` or similar?

File: signer/src/main.rs (hunk 1 above)
Comment (djordon) [reply signer/src/main.rs:83]: nope, will update.

File: signer/src/transaction_signer.rs (hunk 2 above)
Comment (djordon) [reply signer/src/transaction_signer.rs:653]: Good point, will update.

Comment (djordon) [reply signer/src/transaction_signer.rs]: Done in https://github.com/stacks-network/sbtc/pull/589/commits/59c41005ab2e6dca751f18aa1ea960129e28acc9.

**Pull request merged**